### PR TITLE
CI: support PT 1.10

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding support for PT 1.10 nightly

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

File: dockers/base-conda/Dockerfile [hunk 1]
```diff
@@ -111,18 +114,24 @@ RUN \
...
+    if [ $py_ver -eq "0" ]; then \
+        pip install --extra-index-url https://developer.download.nvidia.com/compute/redist "nvidia-dali-cuda${CUDA_VERSION_MAJOR}0>1.0" ; \
+        python -c 'from nvidia.dali.pipeline import Pipeline' ; \
+    fi
 
 RUN \
     # install NVIDIA apex
-    # TODO: later commits break CI when cpp extensions are compiling. Unset when fixed
-    pip install --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" git+https://github.com/NVIDIA/apex@705cba9
+    pip install --no-cache-dir --global-option="--cuda_ext" https://github.com/NVIDIA/apex/archive/refs/heads/master.zip && \
```
Comment (SeanNaren) [inline dockers/base-conda/Dockerfile:128]: You sure we want to point to master for APEX? just hesitant as I'd prefer stability over latest changes on the deprecated package

File: .github/workflows/ci_dockers.yml [hunk 2]
```diff
@@ -62,64 +62,55 @@ jobs:
...
-          - python_version: 3.9
-            pytorch_version: 1.9
+          - python_version: "3.6"
+            pytorch_version: "1.5"
+          - python_version: "3.7"
+            pytorch_version: "1.6"
+          - python_version: "3.8"
+            pytorch_version: "1.9"
+          - python_version: "3.9"
+            pytorch_version: "1.10"
```
Comment (SeanNaren) [inline .github/workflows/ci_dockers.yml:106]: Just out of my curiosity, what's happening here? i see a few versions missing which I'm assuming is intentional

Comment (Borda) [reply .github/workflows/ci_dockers.yml:106]: the CI is testing randomly here, we don't test all 60 combinations

File: dockers/base-conda/Dockerfile (hunk 1 above)
Comment (Borda) [reply dockers/base-conda/Dockerfile:128]: at least actual master is fine, we can freeze it later if needed :]

**Review by carmocca — APPROVED** [review]

**Review by edenlightning — APPROVED** [review]

**Comment by awaelchli** [comment]
@Borda why did you remove "collect_env_details"? 
It was useful to collect details for our bug report template

**Comment by Borda** [comment]
> @Borda why did you remove `tests/collect_env_details.py`?
> It was useful to collect details for our bug report template

it was just moved to the requirements folder and all links to it shall be also updated...

**Comment by awaelchli** [comment]
okay, you missed a few places if we grep tests/collect_env_details.py

**Pull request merged**